pandas/core/indexes/base.py

```diff
@@ -5200,13 +5200,14 @@ def _maybe_cast_slice_bound(self, label, side: str_t, kind):
         # We are a plain index here (sub-class override this method if they
         # wish to have special treatment for floats/ints, e.g. Float64Index and
         # datetimelike Indexes
-        # reject them
-        if is_float(label):
+        # reject them, if index does not contain label
+        if is_float(label) and label not in self.values:
```

Comment:
looks like these can be the same case now
```if (is_float(label) or is_integer(label)) and label not in self.values```

Reply:
Done, thx